endpoints/v2/manifest.py

```diff
@@ -142,7 +142,11 @@ def fetch_manifest_by_digest(namespace_name, repo_name, manifest_ref, registry_m
 
     try:
         manifest = registry_model.lookup_manifest_by_digest(
-            repository_ref, manifest_ref, raise_on_error=True
+            repository_ref,
+            manifest_ref,
+            raise_on_error=True,
+            allow_dead=True,
+            require_available=True,
```

Comment:
allow_dead isn't exactly what you need. It'll return expired manifests, which is incorrect. Quay should always pretend that they are deleted.

Why do you need `require_available`? What happens if `require_available` isn't set?